cmd/fits-api/maps.go

```diff
@@ -13,172 +15,173 @@ type st struct {
 	siteID string
 }
 
-func siteMap(r *http.Request, h http.Header, b *bytes.Buffer) *weft.Result {
-	if res := weft.CheckQuery(r, []string{}, []string{"networkID", "siteID", "sites", "width", "bbox", "insetBbox"}); !res.Ok {
-		return res
+func siteMap(r *http.Request, h http.Header, b *bytes.Buffer) error {
+	q, err := weft.CheckQueryValid(r, []string{"GET"}, []string{}, []string{"networkID", "siteID", "sites", "width", "bbox", "insetBbox"}, valid.Query)
+	if err != nil {
+		return err
 	}
-	h.Set("Content-Type", "image/svg+xml")
-
-	v := r.URL.Query()
 
-	bbox := v.Get("bbox")
-
-	var insetBbox string
-
-	if v.Get("insetBbox") != "" {
-		insetBbox = v.Get("insetBbox")
+	h.Set("Content-Type", "image/svg+xml")
 
-		err := map180.ValidBbox(insetBbox)
+	if q.Get("insetBbox") != "" {
```

Comment:
map180.ValidBbox deals with empty string. No need to check empty parameter here.